src/client/java/net/neoforged/neoforge/client/event/ConfigureGpuDeviceEvent.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) Forge Development LLC and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.client.event;
+
+import net.neoforged.bus.api.Event;
+import net.neoforged.fml.event.IModBusEvent;
+import net.neoforged.neoforge.client.neo3d.GpuDeviceFeatures;
+import net.neoforged.neoforge.client.neo3d.GpuDeviceProperties;
+
+public class ConfigureGpuDeviceEvent extends Event implements IModBusEvent, GpuDeviceFeatures {
```

Comment:
Implementing `GpuDeviceFeatures` is kinda useless here.

Reply:
Allows it to be passed to the record constructor, this is done when making the record kept for the device lifetime (and returned by enabledFeatures) off of the ones enabled on the event.